system/jlib/jsocket.cpp

```diff
@@ -368,6 +368,73 @@ enum SOCKETMODE { sm_tcp_server, sm_tcp, sm_udp_server, sm_udp, sm_multicast_ser
 # endif
 #endif
 
+static CriticalSection queryKACS;
+
+enum UseKA { UNINIT, DISABLED, ENABLED };
+static std::atomic<UseKA> doKeepAlive { UNINIT };
+static int keepAliveTime = -1;
+static int keepAliveInterval = -1;
+static int keepAliveProbes = -1;
+
+/*
+<Software>
+  <Globals>
+    <expert>
+      <keepalive time="200" interval="75" probes="9"/>
+    </expert>
+  </Globals>
+
+global:
+  expert:
+    keepalive:
+      time: 200
+      interval: 75
+      probes: 9
+*/
+
+static bool queryKeepAlive(int &time, int &intvl, int &probes)
+{
+    CriticalBlock block(queryKACS);
+
+    if (doKeepAlive == UNINIT)
+    {
+        doKeepAlive = DISABLED;
+
+#ifdef _CONTAINERIZED
+        Owned<IPropertyTree> expert = getGlobalConfigSP()->getPropTree("expert");
+#else
+        // MCK - without this many components will not have a global prop
+        IPropertyTree *expert = nullptr;
+        Owned<IPropertyTree> envtree = getHPCCEnvironment();
+        if (envtree)
+            expert = envtree->queryPropTree("Software/Globals/expert");
```

Comment:
see comment (https://github.com/hpcc-systems/HPCC-Platform/pull/16936/files#r1100062627).
This should be:
```
expert = envtree->queryPropTree("Software/Globals");
```